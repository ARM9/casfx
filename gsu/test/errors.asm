    .include "../casfx.inc"
    .code
main:
    cache war, 1
    cache 1

    fmult r4
    fmult r1, r1
    fmult r1, r1, r1

    lmult r4
    lmult r1, r4
    lmult r1, r4, r1
    lmult r1, r4, r1, r1
    lmult r4, r1, r1, r6

    merge r0, r7
    merge r0, r1, r2
    merge r0, r7, r2
    merge r0, r8, r7

    plot r1
    plot r2
    plot r2, r1
