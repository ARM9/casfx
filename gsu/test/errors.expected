errors.asm(4): Error: Invalid argument(s) for `CACHE`, expected none or `warn`
../casfx.inc(535): Note: Macro was defined here
errors.asm(5): Error: Invalid argument(s) for `CACHE`, expected none or `warn`
../casfx.inc(535): Note: Macro was defined here
errors.asm(7): Error: Invalid argument(s) for FMULT: destination register can't be R4
../casfx.inc(651): Note: Macro was defined here
errors.asm(8): Error: Invalid argument(s) for FMULT: at least one source register must be R6
../casfx.inc(684): Note: Macro was defined here
errors.asm(9): Error: Invalid argument(s) for FMULT: at least one source register must be R6
../casfx.inc(654): Note: Macro was defined here
errors.asm(11): Error: Invalid argument(s) for LMULT: high destination register can't be R4
../casfx.inc(704): Note: Macro was defined here
errors.asm(12): Error: Invalid argument(s) for LMULT: 0, 1, or 4 operands please
../casfx.inc(701): Note: Macro was defined here
errors.asm(13): Error: Invalid argument(s) for LMULT: 0, 1, or 4 operands please
../casfx.inc(701): Note: Macro was defined here
errors.asm(14): Error: Invalid argument(s) for LMULT: at least one source register must be R6
../casfx.inc(735): Note: Macro was defined here
errors.asm(15): Error: Invalid argument(s) for LMULT: high destination register can't be R4
../casfx.inc(704): Note: Macro was defined here
errors.asm(15): Error: Invalid argument(s) for LMULT: low destination register must be R4
../casfx.inc(706): Note: Macro was defined here
errors.asm(17): Error: Invalid argument(s) for MERGE: 0, 1, or 3 operands please
../casfx.inc(752): Note: Macro was defined here
errors.asm(18): Error: Invalid argument(s) for MERGE: source registers must be r7, r8
../casfx.inc(750): Note: Macro was defined here
errors.asm(19): Error: Invalid argument(s) for MERGE: source registers must be r7, r8
../casfx.inc(750): Note: Macro was defined here
errors.asm(20): Error: Invalid argument(s) for MERGE: source registers must be r7, r8
../casfx.inc(750): Note: Macro was defined here
errors.asm(22): Error: Invalid argument(s) for PLOT: 0 or 2 operands (r1, r2)
../casfx.inc(767): Note: Macro was defined here
errors.asm(23): Error: Invalid argument(s) for PLOT: 0 or 2 operands (r1, r2)
../casfx.inc(767): Note: Macro was defined here
errors.asm(24): Error: Invalid argument(s) for PLOT: operands must be r1, r2
../casfx.inc(765): Note: Macro was defined here
